pkg/metrics/process/snapshot_linux.go

```diff
@@ -96,14 +104,56 @@ func (pw *linuxProcess) Pid() int32 {
 func (pw *linuxProcess) Username() (string, error) {
 	var err error
 	if pw.user == "" { // caching user
+		// try to get it from gopsutil and return it if ok
 		pw.user, err = pw.process.Username()
+		if err == nil {
+			return pw.user, nil
+		}
+
+		// get the uid to be retrieved from getent
+		uid, err := pw.uid()
+		if err != nil {
+			return "", err
+		}
+
+		// try to get it using getent
+		pw.user, err = usernameFromGetent(uid)
 		if err != nil {
 			return "", err
 		}
 	}
 	return pw.user, nil
 }
 
+func (pw *linuxProcess) uid() (string, error) {
+	uuids, err := pw.process.Uids()
+	if err != nil {
+		return "", fmt.Errorf("error getting process uids: %w", err) //nolint:wrapcheck
+	}
+
+	if len(uuids) == 0 {
+		return "", errInvalidUidsForProcess //nolint:wrapcheck
+	}
+
+	return strconv.Itoa(int(uuids[0])), nil
```

Comment:
Just a nitpick, wdyt about returning the actual numeric representation here and do the conversion inside `usernameForGetent`? Returning a string representation could make other usages of this function unwieldy (figure a function that accepts some uid argument and needs to get the number back from the string repr before actually working with the value 😕 )

Reply:
Good point 👍 . Changed in 8325d8796375b02d87ad50bbf2655fa3a7f05e44